client/packages/system/src/Stock/api/hooks/useStockLine.ts

```diff
@@ -15,76 +16,185 @@ export interface DraftStockLine extends StockLineRowFragment {
   inventoryAdjustmentReason: InventoryAdjustmentReasonRowFragment | null;
 }
 
-export function useStockLine() {
-  const [draft, setDraft] = useState<DraftStockLine>({
-    __typename: 'StockLineNode',
-    id: '',
-    itemId: '',
-    onHold: false,
-    packSize: 0,
-    sellPricePerPack: 0,
-    costPricePerPack: 0,
-    totalNumberOfPacks: 0,
-    availableNumberOfPacks: 0,
-    storeId: '',
-    item: {
-      __typename: 'ItemNode',
-      code: '',
-      name: '',
-    },
-    inventoryAdjustmentReason: null,
-  });
+const defaultDraftStockLine = {
+  __typename: 'StockLineNode',
+  id: '',
+  itemId: '',
+  onHold: false,
+  packSize: 0,
+  sellPricePerPack: 0,
+  costPricePerPack: 0,
+  totalNumberOfPacks: 0,
+  availableNumberOfPacks: 0,
+  storeId: '',
+  item: {
+    __typename: 'ItemNode',
+    code: '',
+    name: '',
+  },
+  inventoryAdjustmentReason: null,
+};
+
+export function useStockLine(id?: string) {
+  const [patch, setPatch] = useState<Partial<DraftStockLine>>({});
+  const [isDirty, setIsDirty] = useState(false);
+  const { data, isLoading, error } = useGet(id ?? '');
+  const {
+    mutateAsync: createMutation,
+    isLoading: isCreating,
+    error: createError,
+  } = useCreate();
+  const {
+    mutateAsync: updateMutation,
+    isLoading: isUpdating,
+    error: updateError,
+  } = useUpdate(id ?? '');
+
+  const draft = (
+    data
+      ? { ...data?.nodes[0], ...patch }
+      : { ...defaultDraftStockLine, ...patch }
+  ) as DraftStockLine;
 
-  const { mutateAsync: createMutation } = useCreate();
+  const updatePatch = (newData: Partial<DraftStockLine>) => {
+    // Only add changed values to patch
+    const changedData = data
+      ? Object.fromEntries(
+          Object.entries(newData).filter(
+            ([key, value]) => value !== draft[key as keyof DraftStockLine]
```

Comment:
Was this needed because of calling updatePatch with the full object somewhere? Would have thought just updating the patch with the `newData` would have been fine??

Reply:
It prevents that same problem you dealt with the other day -- clicking and leaving an input triggers `onChange`, so you get the buttons becoming enabled even when you haven't actually changed anything. 

So we just check for actual changed values and filter out the ones that aren't.

Reply:
Hi - I think this issue should be solved in the `CurrencyInput` component (can be another issue/PR), or at least in the `StockLineForm` rather than adding more logic to these API hooks to compensate... should just be able to do `setPatch({ ...patch, newData })` here?

Reply:
Okay, have changed it a bit, so now it'll compare the current patch state with the saved data and toggle `isDirty` accordingly (if actually different):

![2024-06-04 11 48 08](https://github.com/msupply-foundation/open-msupply/assets/5456533/3e32ee24-918c-49e5-8c49-904d352b699b)

f4e1bf21b